### ISSUE TITLE
"internal server error" but nothing in logs or traces

### ISSUE THREAD

**Issue by nathanael-h** [freifunkMUC/wg-access-server#179]
Hello, 
I have an issue similar to https://github.com/freifunkMUC/wg-access-server/issues/154 but I am not certain it is related to auth. 

I am running the v0.0.7 with docker-compose. The only auth mechanism enabled is OIDC. Sometimes users got a 500 error with "internal server error" on the webui. This only happens if the user has already logged-in at least once. If he cleans cookies and reload it works (redirect to the OIDC, then back to the wg-access-server webui). 

I enabled `trace` to have some logs, but unfortunately I see nothing in the container output about this. 

Here is a redacted request that give 500, both in Firefox and curl: 
```
curl -iL 'https://vpn.example.com/'  -H 'DNT: 1' -H 'Cookie: rl_anonymous_id=%224b7b75b9-xxxx-xxxx-xxxx-133c721ed249%22; rl_user_id=%22%22; auth-session=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'

HTTP/2 500 
server: nginx
date: Mon, 16 May 2022 09:33:43 GMT
content-type: text/plain; charset=utf-8
content-length: 22
x-content-type-options: nosniff

internal server error

```

Looking for that in the code I found that logs should come from there, isn'it?  https://github.com/freifunkMUC/wg-access-server/blob/b5992feecdeb8433fcfc75b4fd3e53255f3290df/pkg/authnz/router.go#L103

Is there something missing for an error being caught and outputted in logs? I tried to look for that, unsuccessfully.

**Comment by DasSkelett** [comment]
Ugh, that's the problem with the `ctxlogrus` being used in places it shouldn't be and thus being a no-op. We'll have to replace it with a proper logger there.

But looking at the line, the only situation where there can be an error is if `session.Identity`/`user` is nil making `claimsMiddleware` return an error:
https://github.com/freifunkMUC/wg-access-server/blob/b5992feecdeb8433fcfc75b4fd3e53255f3290df/cmd/serve/main.go#L355-L365

Not sure how this can happen for users that _should_ be logged in.

**Comment by nathanael-h** [comment]
About replacing `ctxlogrus` I think I won't be able to provide PR or help... 

I will work on reproducing the issue, and checking what is happening on the OIDC backend server we use. Maybe the real problem in on this server. I would have use logs, but your comment above help me also.

**Comment by nathanael-h** [comment]
Nope, I did not took the time to dig further on my OIDC provider.
 No problem for the time 😉 Thanks for your PR #194 I will try this hopefully in the next days. And update you.

**Comment by DasSkelett** [comment]
I can trigger this error by aborting the login on the OIDC provider's page and opening any wg-access-server but /signin again.
This is because we don't set an identity in the loginHandler for OIDC, because there is none yet, until the callback.
https://github.com/freifunkMUC/wg-access-server/blob/29c783fedb5214e63b7ea2b8522d763606e417bb/pkg/authnz/authconfig/oidc.go#L75-L77

As far as I have seen this is the only place where we create an `AuthSession` object without `Identitiy`, so I'm optimistic that a failed OIDC login is the cause of your user reports as well.

For this issue simply redirecting to the signin page again should be the best course of actions, making the user log into the OIDC provider again, hopefully this time successfully.

**Comment by nathanael-h** [comment]
Hello @DasSkelett you were faster than me: we made some tests and looked in the code yesterday, and we found like you that the issue `User is not logged in` is probably related to a `AuthSession` without `Identity`. 
Like you we could reproduce the issue by aborting the OIDC login flow, and coming back to the homepage. I wanted to report it, but you have already filled issue #201 and merge a PR to fix it in #202 :clap:  :clap:  :clap: 

There is another pattern which leads to this same 500 error. Someone login via OIDC, it works. The same person come back later (generally what I saw was a few days) and there this same 500 error. We were asking ourselves if it was possible to *loose* the  `AuthSession.Identity`... but this seems very strange. We looked in the code about `AuthSession`, `Store` but found nothing relevant. 

I will build and run your latest changes, and update you. Thanks again!

**Comment by nathanael-h** [comment]
I confirm #202 fixes the 500 internal server error that raised when OIDC login is aborted. Thanks!
Do you have an ETA for a new release?

**Comment by DasSkelett** [comment]
Nice! We already published v0.8.0 yesterday with this fix included: https://github.com/freifunkMUC/wg-access-server/releases/tag/v0.8.0

**Comment by nathanael-h** [comment]
Oops :hand_over_mouth:
